src/table.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const React = require('react');
+const mdl = require('material-design-lite/material.min');
+const classnames = require('classnames');
+
+const baseClasses = {
+  'mdl-data-table': true,
+  'mdl-js-data-table': true
+};
+
+class Table extends React.Component {
+
+  componentDidMount(){
+    const node = React.findDOMNode(this);
+    mdl.upgradeElement(node);
+  }
+
+  componentWillUnmount(){
+    const node = React.findDOMNode(this);
+    mdl.downgradeElements(node);
+  }
+
+  render(){
+    const {
+      className,
+      columns,
+      rows,
+      selectable,
+      numeric
+    } = this.props;
+
+    const classes = classnames(baseClasses, {
+      'mdl-data-table--selectable': selectable
+    }, className);
+
+    const cellClasses = classnames({
+      'mdl-data-table__cell--non-numeric': numeric
+    });
+
+    return (
+      <table {...this.props} className={classes}>
+        <thead>
+          <tr>
+            {
+              columns.map((column) => {
```

Comment:
prefer this to be assigned to a `const` above or use single line arrow functions with implicit return